flink-runtime/src/test/java/org/apache/flink/runtime/blob/BlobServerPutTest.java

```diff
@@ -59,6 +60,110 @@
 
 	private final Random rnd = new Random();
 
+
+	// --- concurrency tests for utility methods which could fail during the put operation ---
+
+	/**
+	 * Checked thread that calls {@link BlobServer#getStorageLocation(BlobKey)}
+	 */
+	public static class ContentAddressableGetStorageLocation extends CheckedThread {
+		private final BlobServer server;
+		private final BlobKey key;
+
+		public ContentAddressableGetStorageLocation(BlobServer server, BlobKey key) {
+			this.server = server;
+			this.key = key;
+		}
+
+		@Override
+		public void go() throws Exception {
+			server.getStorageLocation(key);
```

Comment:
Unfortunately, concurrency with `delete` operations does not work either if not guarded - the directory may not exist anymore between `jobDirectory.mkdirs()` and `jobDirectory.exists()`. I was able to reproduce the error with the existing test though - if you want to try it, just change the order of these two commands back - the test will not hit every time, but some times.